rust/crd/src/security.rs

```diff
@@ -224,7 +224,7 @@ impl KafkaTlsSecurity {
 
         // When users enable Kerberos we require them to also enable TLS for maximum security and
         // to limit the number of combinations we need to support.
-        if self.has_kerberos_enabled() || self.tls_client_authentication_class().is_some() {
+        if self.has_kerberos_enabled() {
```

Comment:
Did we need to change this because something was broken? I would have expected that it's legit to say "when you enable mutual tls, you also need to enable tls"
If I'm right, we would either need to reword the error message or have to different errors and ensure!()

Reply:
One of the upgrade test fails because it sets internal to tls but serverSecret is null.

Reply:
I'll have another look and see which one it was.

Reply:
Yes, this is expected behaviour but we can re-visit this in a separate [issue](https://github.com/stackabletech/kafka-operator/issues/767).

Reply:
Thanks! Happy to resolve the thread